### PR TITLE
NIAD-1188: changing default mailbox and fixing issue #106

### PR DESCRIPTION
**Pull request by bartek-sarul**
Fixing issue: https://github.com/nhsconnect/integration-adaptor-lab-results/issues/106

## Description

Please explain the changes you have made here.

## Jira Ticket

Please add a link to the related Jira ticket (if any).

## Checklist

These are items (excluding GitHub Checks) which should be confirmed before a branch is ready to merge.

- [ ] Acceptance Criteria met
- [ ] Commit messages are meaningful
- [ ] Manually tested
- [ ] Self-reviewed your code
- [ ] Code reviewed from two other developers
- [ ] If your pull request depends on any other, please link them in the description
- [ ] main branch is passing/green on CI
- [ ] Add/update any relevant documentation

### CODE REVIEW

**Comment by saileshp-kainos** [comment]
There's been a new mailbox ID recipient (`110000044300001`) added in the `main` branch so we need the `main` branch merged in

File: vnp/README.md [hunk 1]
```diff
@@ -14,9 +14,9 @@ Use the MESH Client script to send the same file multiple times to a given MESH
 
 Example:
 
-    ./mesh.sh send gp_mailbox "@../src/intTest/resources/edifact/pathology.edifact.dat" 3 
+    for ((i = 1 ; i <= 3; i++)); do ./mesh.sh send lab_results_mailbox "@../src/intTest/resources/edifact/pathology.edifact.dat"; done
```
Comment (andrewwa-kainos) [inline vnp/README.md:17]: This file was renamed to `pathology_3.edifact.dat` (with a `_2` counterpart)

Comment (bartek-sarul) [reply vnp/README.md:17]: I'll update with main branch and check

File: docker-compose.yml [hunk 2]
```diff
@@ -119,12 +119,12 @@ services:
...
-          000000183500005=lab_results_mailbox
-      - LAB_RESULTS_MESH_POLLING_CYCLE_MINIMUM_INTERVAL_IN_SECONDS=300
-      - LAB_RESULTS_MESH_POLLING_CYCLE_DURATION_IN_SECONDS=285
-      - LAB_RESULTS_MESH_CLIENT_WAKEUP_INTERVAL_IN_MILLISECONDS=60000
+      - LAB_RESULTS_MESH_RECIPIENT_MAILBOX_ID_MAPPINGS=000000004400001=lab_results_ack_mailbox
+          000000024600002=lab_results_ack_mailbox
+          000000183500005=lab_results_ack_mailbox
+      - LAB_RESULTS_MESH_POLLING_CYCLE_MINIMUM_INTERVAL_IN_SECONDS=3
+      - LAB_RESULTS_MESH_POLLING_CYCLE_DURATION_IN_SECONDS=5
+      - LAB_RESULTS_MESH_CLIENT_WAKEUP_INTERVAL_IN_MILLISECONDS=10000
```
Comment (matthewdoreilly) [inline docker-compose.yml:128]: So we'll scan the MESH mailbox for new messages every 10 seconds (changed from 60 seconds). What's the reason for this change? Is this a new requirement from the business for performance of the adaptor?

File: docker-compose.yml [hunk 3]
```diff
@@ -119,13 +119,13 @@ services:
...
         3WJ4bVSmp3MOQ2EF4zrAl4fXEd4lS3JxKtzbHFbSogAmPFnuP8wIDWTDgfEq
         -----END RSA PRIVATE KEY-----
-      - LAB_RESULTS_MESH_RECIPIENT_MAILBOX_ID_MAPPINGS=000000004400001=lab_results_mailbox
-          000000024600002=lab_results_mailbox
-          000000183500005=lab_results_mailbox
-          110000044300001=lab_results_mailbox
-      - LAB_RESULTS_MESH_POLLING_CYCLE_MINIMUM_INTERVAL_IN_SECONDS=300
-      - LAB_RESULTS_MESH_POLLING_CYCLE_DURATION_IN_SECONDS=285
-      - LAB_RESULTS_MESH_CLIENT_WAKEUP_INTERVAL_IN_MILLISECONDS=60000
+      - LAB_RESULTS_MESH_RECIPIENT_MAILBOX_ID_MAPPINGS=000000004400001=lab_results_ack_mailbox
```
Comment (matthewdoreilly) [inline docker-compose.yml:122]: Why this change? Shouldn't this stay as `lab_results_mailbox`?

File: mesh/mesh.sh [hunk 4]
```diff
@@ -28,44 +26,39 @@ create_token() {
...
+    --cert "${MESH_ENDPOINT_CERT}" --key "${MESH_ENDPOINT_PRIVATE_KEY}" -H "Authorization: ${TOKEN}" \
     -H 'Mex-ClientVersion: 1.0' -H 'Mex-JavaVersion: 1.7.0_60' -H 'Mex-OSArchitecture: Windows 7' -H 'Mex-OSName: x86' -H 'Mex-OSVersion: 6.1'
 }
 
 send() {
   local body
-  local count
   body="$1"
-  count="${2:-1}"
-  for ((i = 1 ; i <= $count ; i++)); do
```
Comment (matthewdoreilly) [inline mesh/mesh.sh:46]: This is required for the performance testing, how come you're reverting that?

**Review by matthewdoreilly — COMMENTED** [review]
Looks fine to me, my comments are mainly just wondering why some of these changes are being made

File: docker-compose.yml (hunk 2 above)
Comment (bartek-sarul) [reply docker-compose.yml:128]: That's only for the docker-compose test environment to speed up testing.

File: docker-compose.yml (hunk 3 above)
Comment (bartek-sarul) [reply docker-compose.yml:122]: That's for testing purpose only. Having the same mailbox for both interchanges and nhsacks adds a lot of noise when testing.

File: mesh/mesh.sh (hunk 4 above)
Comment (bartek-sarul) [reply mesh/mesh.sh:46]: Fro some reason a loop inside the script didnt work. 1st request pass and the rest fails. Dunno why. So I've moved if outside and described in the vnp readme

Comment (andrewwa-kainos) [reply mesh/mesh.sh:46]: Probably because the token is single-use. You'd need to create a new token for each curl

File: docker-compose.yml (hunk 2 above)
Comment (matthewdoreilly) [reply docker-compose.yml:128]: Ah sorry thought this was the app config (and not docker), yeah that's fine

File: vnp/README.md (hunk 1 above)
Comment (saileshp-kainos) [reply vnp/README.md:17]: Yeah, `pathology.dat` doesn't exist anymore as we replace it with `pathology_2.edifact.dat` and `pathology_3.edifact.dat` for NHS002 and NHS003 messages.

File: src/main/resources/application.yml [hunk 5]
```diff
@@ -132,10 +132,10 @@ labresults:
       3WJ4bVSmp3MOQ2EF4zrAl4fXEd4lS3JxKtzbHFbSogAmPFnuP8wIDWTDgfEq
       -----END RSA PRIVATE KEY-----}
     subCAcert: ${LAB_RESULTS_MESH_SUB_CA:}
-    recipientToMailboxIdMappings: ${LAB_RESULTS_MESH_RECIPIENT_MAILBOX_ID_MAPPINGS:000000004400001=lab_results_mailbox
-      000000024600002=lab_results_mailbox
-      000000183500005=lab_results_mailbox
-      110000044300001=lab_results_mailbox}
+    recipientToMailboxIdMappings: ${LAB_RESULTS_MESH_RECIPIENT_MAILBOX_ID_MAPPINGS:000000004400001=lab_results_ack_mailbox
```
Comment (matthewdoreilly) [inline src/main/resources/application.yml:135]: The test mailbox name from docker-compose (`lab_results_ack_mailbox`) will be passed here when testing locally, so shouldn't we just keep the default mailbox name still as `lab_results_mailbox` here?
Like we do for `wakeupIntervalInMilliseconds` which is "60000" by default

Comment (bartek-sarul) [reply src/main/resources/application.yml:135]: In real life "our" inbox will in fact be different than the one we send nhsacks to (owned by the lab).
Let's leave the other defaults in the app as they are as there is a chance the test partners don't alter that when deploying to prod. This will be a breaking change for them. On the other hand `recipientToMailboxIdMappings` is something that they have to change in order to make prod to work.

File: vnp/README.md [hunk 6]
```diff
@@ -14,16 +14,28 @@ Use the MESH Client script to send the same file multiple times to a given MESH
...
 
 ## Sending EDIFACT to `Inbound MESH Queue`
 
 Use the ["A" application](https://github.com/fmtn/a) which is an AMQP client allowing you to send messages directly to `Inbound MESH Queue`. Java 11 required.
 
+Wrap any Med Report EDIFACT message into json (example: [pathology_3.edifact.dat](`../src/intTest/resources/edifact/pathology_3.edifact.dat`)) as follows:
+```json
+{
+    "recipient": null,
+    "workflowId": "SCRN_BCS_MEDRPT_V4",
```
Comment (saileshp-kainos) [inline vnp/README.md:29]: Because in the example, we mention `pathology_3.edifact.dat`

```suggestion
    "workflowId": "PATH_MEDRPT_V3",
```

**Review by andrewwa-kainos — DISMISSED** [review]

**Review by saileshp-kainos — APPROVED** [review]

**Review by andrewwa-kainos — APPROVED** [review]

**Review by matthewdoreilly — APPROVED** [review]

**Review by antonlebedjko — APPROVED** [review]
